documentation/bolt_examples.md

```diff
@@ -123,3 +123,228 @@ To convert the `restart_service.ps1` script to a task, giving you the ability to
+```
+
+5. To make sure that your inventory is configured correctly and that you can connect to all the target nodes, run the following command from inside the project directory: 
+
+```
+bolt command run 'echo hi' --targets windows
+```
+
+**Note:** The `--targets windows` argument refers to the node group defined in the inventory file.
+
+You should get the following output:
+
+```
+Started on x.x.x.x...
+Started on localhost...
+Finished on localhost:
+  STDOUT:
+    hi
+Finished on 0.0.0.0:
+  STDOUT:
+    hi
+Finished on 127.0.0.1:
+  STDOUT:
+    hi
+Successful on 3 nodes: 0.0.0.0:20022,localhost
+Ran on 3 nodes in 0.20 seconds
+```
+
+### 2. Download the Chocolatey module and its dependencies from the Puppet Forge. 
+
+Bolt uses a `Puppetfile` to install module content from the Forge. A `Puppetfile` is a formatted text file that specifies the modules and data you want in each environment.
+
+1. Create a file named `Puppetfile` in the project directory — which describes the Forge content you want to install: 
+
+```
+forge 'https://forge.puppet.com'
+        
+        # Modules from the Puppet Forge
+        mod 'puppetlabs-chocolatey', '4.1.0'
+        ## dependencies
+        mod 'puppetlabs-stdlib', '4.13.1' #install latest
+        mod 'puppetlabs-powershell', '2.3.0'
+        mod 'puppetlabs-registry', '2.1.0'
+        
+        # Modules from Git
+        # Examples: https://github.com/puppetlabs/r10k/blob/master/doc/puppetfile.mkd#examples
+        #mod 'apache',
+        #  :git    => 'https://github.com/puppetlabs/puppetlabs-apache',
+        #  :commit => 'de290646f97e04b4b8e42c70f6e01e860c394ce7'     
+ ```
+
+2. Add the Puppetfile to a version control service, for example, Github.
+
+3. Add the following code to the Puppetfile — with your git user account: 
+
+```     
+mod 'puppet_choco_tap',
+    :git    =>  'https://github.com/<your git user>/puppet_choco_tap.git'
+```
+
+4. From inside the project directory, install the Forge content by running: 
+
+```
+bolt puppetfile install
+```
+
+After it runs, you can see a `modules` directory inside the project directory, containing the modules you specified in the `Puppetfile`.
+
+### 3. Write a Bolt plan to apply Puppet code and orchestrate the deployment of a package resource using the Chocolatey provider. Plans allow you to run more than one task with a single command.
+
+1. Inside your project directory, in a file called `/plan/installer.pp`, create a plan called `puppet_chocolatey_tap::installer`. 
+
+The folder tree should look like this:
+
+```
+├──plan
+   └── installer.pp 
+```
+
+2. Copy the following code into the `installer.pp` file:
+
+```
+plan puppet_choco_tap::installer(
+   TargetSpec                                   $nodes,
+   String                                       $package,
+   Variant[Enum['absent', 'present'], String ]  $ensure = 'present',
+ ){
+
+ apply_prep($nodes)
+
+apply($nodes){
+  include chocolatey
+ }
+apply($nodes){
```

Comment:
The indentations for the second apply block are funky. Also, I think that given we do not do anything with the result of the first apply statement and apply the manifest code on the same sets of targets, I think we should have a single statement with all the manifest code (bolt performance).

Reply:
@donoghuc I'm not quite sure what you mean, would you mind updating the code with this?

Reply:
```puppet
plan puppet_choco_tap::installer(
  TargetSpec $nodes,
  String $package,
  Variant[Enum['absent', 'present'], String ] $ensure = 'present',
){
  apply_prep($nodes)

  apply($nodes){
    include chocolatey

    package { $package :
      ensure    => $ensure,
      provider  => 'chocolatey',
      }
    }
  }
}
```